### DISCUSSION TITLE
How to hidde password on body request

### DISCUSSION THREAD

**Discussion by Allorentee** [supabase/supabase#20223] [Questions] [answered]
Good, in the method of registration or login by mail is passed the password in the body of the request and I think it is not the safest thing in the world, I tried to send it encrypted but when I do the login then does not match the one that has saved supabase, how can I prevent the password is displayed in the body of the request and if I can not encrypt it or some other solution

**Comment by GaryAustin1** [comment]
The body is encrypted automatically by the browser for transit with https protocol.  This is standard method used for almost all apps you encounter that run on browsers.

If you encrypt first and pass that then you have doubly encrypted the password and you would have to have a means to decrypt it on the server.  If you just use the encrypted value as the password that is no different than just type a random string as the original password.

**Reply by Allorentee** [reply]
But im acBut now I am making the requests from a web with https protocol and even so I keep getting the passwords in the body of the requests.
<img width="1531" alt="Captura de pantalla 2024-01-07 a las 22 41 28" src="https://github.com/supabase/supabase/assets/99037086/934b54ca-49a0-4b9d-9e7c-68069ba2c6a9">

**Reply by GaryAustin1** [reply, marked as answer]
Yes.  You can see the body in the browser console.  But when it leaves the browser onto the net the body will be encrypted.  No reason the user should not able to see their own password if they use dev mode.
https://stackoverflow.com/questions/21514890/password-sent-via-https-accessible-via-browser-developer-tools
https://stackoverflow.com/questions/16306839/should-data-in-an-https-request-appear-as-encrypted-in-chrome-developer-tools

Feel free to do other research on the web about safety of https for passwords, but is not an issue.

**Reply by Allorentee** [reply]
Okey, thank you very much for the information.
